File: tool/src/main/java/com/glodon/tool/SharedPreferencesUtil.java
package com.glodon.tool;

import android.content.Context;
import android.content.SharedPreferences;

import java.util.Set;

public class SharedPreferencesUtil {
    private String KEY = "com.glodon.glm.sharePrefer";
    private SharedPreferences mSharedPreferences;
    private static SharedPreferencesUtil _instance = new SharedPreferencesUtil();
    public static SharedPreferencesUtil Instance() {
        return _instance;
    }

    public void init(Context context) {
        if (mSharedPreferences == null) {
            mSharedPreferences = context.getApplicationContext().getSharedPreferences(KEY, 0);
        }
    }

    /**
     * str: md5+isTracing
     */
    public void put(String id, String str) {
        mSharedPreferences.edit().putString(id, str).commit();
    }

    public String get(String id) {
        return mSharedPreferences.getString(id, "");
    }

    public String getMd5(String id) {
        String str = mSharedPreferences.getString(id, "");
        if (!"".equals(str)) {
            return str.substring(0, str.length() - 1);
        }
        return "";
    }

    public int getIsTracing(String id) {
        String str = mSharedPreferences.getString(id, "");
        if (str != null && str.length() > 0) {
            return Integer.valueOf(str.substring(str.length() - 1), str.length());
        }
        return 1;
    }

    public void remove(String id) {
        mSharedPreferences.edit().remove(id).commit();
    }

    public Set<String> getIds() {
        return mSharedPreferences.getAll().keySet();
    }

    public void clear() {
        mSharedPreferences.edit().clear().commit();
    }

    public void putBoolean(String key, boolean b) {
        mSharedPreferences.edit().putBoolean(key, b).commit();
    }

    public boolean getBoolean(String key) {
        return mSharedPreferences.getBoolean(key, false);
    }

    public boolean getBooleanDefaultFalse(String key) {
        return mSharedPreferences.getBoolean(key, false);
    }

    public boolean putInt(String mainPort, int port) {
        return mSharedPreferences.edit().putInt(mainPort, port).commit();
    }

    public int getInt(String key) {
        return mSharedPreferences.getInt(key, 0);
    }


    public boolean putFloat(String mainPort, float port) {
        return mSharedPreferences.edit().putFloat(mainPort, port).commit();
    }

    public float getFloat(String key) {
        return mSharedPreferences.getFloat(key, 0);
    }
}

File: kotlintools/src/main/java/com/glodon/glm/kotlintools/SharedPreferencesUtil.kt
package com.glodon.glm.kotlintools

import android.content.Context
import android.content.SharedPreferences

/**
 * Created by lichongmac@163.com on 2020/6/12.
 */
object SharedPreferencesUtil {
    private const val KEY = "com.glodon.glm.sharePrefer"
    private var mSharedPreferences: SharedPreferences? = null

    fun init(context: Context) {
        if (mSharedPreferences == null) {
            mSharedPreferences = context.applicationContext.getSharedPreferences(KEY, 0)
        }
    }

    /**
     * str: md5+isTracing
     */
    fun put(id: String, str: String) {
        mSharedPreferences!!.edit().putString(id, str).commit()
    }

    operator fun get(id: String?): String? {
        return mSharedPreferences!!.getString(id, "")
    }

    fun getMd5(id: String?): String? {
        val str = mSharedPreferences!!.getString(id, "")
        return if ("" != str) {
            str!!.substring(0, str.length - 1)
        } else ""
    }

    fun getIsTracing(id: String?): Int {
        val str = mSharedPreferences!!.getString(id, "")
        return if (str!!.isNotEmpty()) {
            Integer.valueOf(str.substring(str.length - 1), str.length)
        } else 1
    }

    fun remove(id: String?) {
        mSharedPreferences!!.edit().remove(id).commit()
    }

    fun getIds(): Set<String?>? {
        return mSharedPreferences!!.all.keys
    }

    fun clear() {
        mSharedPreferences!!.edit().clear().commit()
    }

    fun getBoolean(key: String?): Boolean {
        return mSharedPreferences!!.getBoolean(key, false)
    }

    fun getBooleanDefaultFalse(key: String?): Boolean {
        return mSharedPreferences!!.getBoolean(key, false)
    }

    fun putInt(mainPort: String?, port: Int): Boolean {
        return mSharedPreferences!!.edit().putInt(mainPort, port).commit()
    }

    fun getInt(key: String?): Int {
        return mSharedPreferences!!.getInt(key, 0)
    }


    fun putFloat(mainPort: String?, port: Float): Boolean {
        return mSharedPreferences!!.edit().putFloat(mainPort, port).commit()
    }

    fun getFloat(key: String?): Float {
        return mSharedPreferences!!.getFloat(key, 0f)
    }


}